homeassistant/components/openweathermap/__init__.py

```diff
@@ -17,24 +18,33 @@
     CONF_MODE,
     CONF_NAME,
 )
-from homeassistant.core import HomeAssistant
+from homeassistant.core import CALLBACK_TYPE, HomeAssistant
 
 from .const import (
     CONFIG_FLOW_VERSION,
-    DOMAIN,
-    ENTRY_NAME,
-    ENTRY_WEATHER_COORDINATOR,
     FORECAST_MODE_FREE_DAILY,
     FORECAST_MODE_ONECALL_DAILY,
     PLATFORMS,
-    UPDATE_LISTENER,
 )
 from .weather_update_coordinator import WeatherUpdateCoordinator
 
 _LOGGER = logging.getLogger(__name__)
 
+OpenweathermapConfigEntry = ConfigEntry["OpenweathermapRuntimeData"]
 
-async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+
+@dataclass
+class OpenweathermapRuntimeData:
```

Comment:
Maybe change the name to just `OpenweathermapData`? Runtime feels a bit redundant / unnecessary here.